File: geophone-plus-2-python-examples/get_phone_Info_rest_sdk_example.py
import sys
import os

sys.path.insert(0, os.path.abspath("../geophone-plus-2-python/REST"))

from get_phone_Info_rest import get_phone_info


def get_phone_info_rest_sdk_go(is_live: bool, license_key: str) -> None:
    print("\n----------------------------------------------")
    print("GeoPhone Plus 2 - GetPhoneInfoInput - REST SDK")
    print("----------------------------------------------")

    phone_number = "8059631700"
    test_type = "FULL"
    timeout_seconds = 15

    print("\n* Input *\n")
    print(f"Phone Number   : {phone_number}")
    print(f"Test Type      : {test_type}")
    print(f"License Key    : {license_key}")
    print(f"Is Live        : {is_live}")
    print(f"Timeout Seconds: {timeout_seconds}")

    try:
        response = get_phone_info(
            phone_number, test_type, license_key, is_live, timeout_seconds
        )

        if response.Error is None:
            print("\n* Phone Info *\n")
            if response.PhoneInfo:
                print(f"SMS Address      : {response.PhoneInfo.SMSAddress}")
                print(f"MMS Address      : {response.PhoneInfo.MMSAddress}")
                print(f"Date First Seen  : {response.PhoneInfo.DateFirstSeen}")
                print(f"Date of Porting  : {response.PhoneInfo.DateOfPorting}")
                print(f"Note Codes       : {response.PhoneInfo.NoteCodes}")
                print(f"Note Descriptions: {response.PhoneInfo.NoteDescriptions}")
                print(f"Tokens Used      : {response.PhoneInfo.TokensUsed}")

                print("\n* Provider Details *\n")
                if response.PhoneInfo.Provider:
                    print(f"Name     : {response.PhoneInfo.Provider.Name}")
                    print(f"City     : {response.PhoneInfo.Provider.City}")
                    print(f"State    : {response.PhoneInfo.Provider.State}")
                    print(f"Line Type: {response.PhoneInfo.Provider.LineType}")
                    print(f"Latitude : {response.PhoneInfo.Provider.Latitude}")
                    print(f"Longitude: {response.PhoneInfo.Provider.Longitude}")
                else:
                    print("No provider details found.")

                print("\n* Contacts *\n")
                if hasattr(response.PhoneInfo, "Contacts") and response.PhoneInfo.Contacts:
                    for contact in response.PhoneInfo.Contacts:
                        print(f"Name           : {contact.Name}")
                        print(f"Address        : {contact.Address}")
                        print(f"City           : {contact.City}")
                        print(f"State          : {contact.State}")
                        print(f"Postal Code    : {contact.PostalCode}")
                        print(f"Phone Type     : {contact.PhoneType}")
                        print(f"Latitude       : {contact.Latitude}")
                        print(f"Longitude      : {contact.Longitude}")
                        print(f"SIC Code       : {contact.SICCode}")
                        print(f"SIC Description: {contact.SICDesc}")
                        print(f"Quality Score  : {contact.QualityScore}")
                        print("")
                else:
                    print("No contacts found.")
            else:
                print("No phone info found.")
        else:
            print("\n* Error *\n")
            print(f"Error Type       : {response.Error.Type}")
            print(f"Error Type Code  : {response.Error.TypeCode}")
            print(f"Error Description: {response.Error.Desc}")
            print(f"Error Desc Code  : {response.Error.DescCode}")

    except Exception as e:
        print(f"\nException occurred: {str(e)}")

File: geophone-plus-2-dot-net-examples/GetPhoneInfoRestSdkExample.cs
using geophone_plus_2_dot_net.REST;
using System;
using System.ComponentModel;
using System.Linq;
using System.Threading.Tasks;

namespace geophone_plus_2_dot_net_examples
{
    public static class GetPhoneInfoRestSdkExample
    {
        public static void Go(string licenseKey, bool isLive)
        {
            Console.WriteLine("\r\n----------------------------------------------");
            Console.WriteLine("GeoPhone Plus 2 - GetPhoneInfoInput - REST SDK");
            Console.WriteLine("----------------------------------------------");

            GetPhoneInfoClient.GetPhoneInfoInput getPhoneInfoInput = new(
                PhoneNumber: "8059631700",
                TestType: "FULL",
                LicenseKey: licenseKey,
                IsLive: isLive,
                TimeoutSeconds: 15);
         
            Console.WriteLine("\r\n* Input *\r\n");
            Console.WriteLine($"Phone Number: {getPhoneInfoInput.PhoneNumber}");
            Console.WriteLine($"Test Type   : {getPhoneInfoInput.TestType}");
            Console.WriteLine($"License Key : {getPhoneInfoInput.LicenseKey}");
            Console.WriteLine($"Is Live     : {getPhoneInfoInput.IsLive}");

            GPPL2Response response = GetPhoneInfoClient.Invoke(getPhoneInfoInput);
            if (response.Error is null)
            {
                if (response.PhoneInfo != null)
                {
                    Console.WriteLine("\r\n* Phone Info *\r\n");
                    Console.WriteLine($"SMS Address    : {response.PhoneInfo.SMSAddress}");
                    Console.WriteLine($"MMS Address    : {response.PhoneInfo.MMSAddress}");
                    Console.WriteLine($"Date First Seen: {response.PhoneInfo.DateFirstSeen}");
                    Console.WriteLine($"Date of Porting: {response.PhoneInfo.DateOfPorting}");
                    Console.WriteLine($"Note Codes     : {response.PhoneInfo.NoteCodes}");
                    Console.WriteLine($"Note Desc      : {response.PhoneInfo.NoteDescriptions}");
                    Console.WriteLine($"Tokens Used    : {response.PhoneInfo.TokensUsed}");

                    Console.WriteLine("\r\n* Provider Details *\r\n");
                    if (response.PhoneInfo.Provider != null)
                    {
                        Console.WriteLine($"Name     : {response.PhoneInfo.Provider.Name}");
                        Console.WriteLine($"City     : {response.PhoneInfo.Provider.City}");
                        Console.WriteLine($"State    : {response.PhoneInfo.Provider.State}");
                        Console.WriteLine($"Line Type: {response.PhoneInfo.Provider.LineType}");
                        Console.WriteLine($"Latitude : {response.PhoneInfo.Provider.Latitude}");
                        Console.WriteLine($"Longitude: {response.PhoneInfo.Provider.Longitude}");
                    }
                    else
                    {
                        Console.WriteLine("No provider details found.");
                    }

                    Console.WriteLine("\r\n* Contacts *\r\n");
                    if (response.PhoneInfo.Contacts?.Length > 0)
                    {
                        foreach (var contact in response.PhoneInfo.Contacts)
                        {
                            Console.WriteLine($"Name         : {contact.Name}");
                            Console.WriteLine($"Address      : {contact.Address}");
                            Console.WriteLine($"City         : {contact.City}");
                            Console.WriteLine($"State        : {contact.State}");
                            Console.WriteLine($"Postal Code  : {contact.PostalCode}");
                            Console.WriteLine($"Phone Type   : {contact.PhoneType}");
                            Console.WriteLine($"Latitude     : {contact.Latitude}");
                            Console.WriteLine($"Longitude    : {contact.Longitude}");
                            Console.WriteLine($"SIC Code     : {contact.SICCode}");
                            Console.WriteLine($"SIC Desc     : {contact.SICDesc}");
                            Console.WriteLine($"Quality Score: {contact.QualityScore}");
                            Console.WriteLine("\r\n");
                        }
                    }
                    else
                    {
                        Console.WriteLine("No contacts found.");
                    }
                }
                else
                {
                    Console.WriteLine("No phone info found.");
                }
            }
            else
            {
                Console.WriteLine("\r\n* Error *\r\n");
                Console.WriteLine($"Error Type       : {response.Error.Type}");
                Console.WriteLine($"Error Type Code  : {response.Error.TypeCode}");
                Console.WriteLine($"Error Description: {response.Error.Desc}");
                Console.WriteLine($"Error Desc Code  : {response.Error.DescCode}");
            }
        }
    }
}